### PR TITLE
setup metering

### PR DESCRIPTION
**Pull request by cah-link**
This integrates the metering pipeline into osism service roles

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.idea
```
Comment (osfrickler) [inline .gitignore:1]: Please don't do this. User-specific ignores belong into user-specific global git configuration, not into projects.

Comment (cah-link) [reply .gitignore:1]: understand. Thank you.

**Comment by berendt** [comment]
@cah-link You have to add a playbook to https://github.com/osism/ansible-playbooks/tree/main/playbooks/infrastructure. ``metering.yml``. Also add a symlink to https://github.com/osism/ansible-playbooks/tree/main/playbooks (``infrastructure-metering.yml``). After we merged this playbook the ``apply`` command will be available.

**Comment by berendt** [comment]
For a test of this branch run ``docker run -it osism-ansible /change.sh cah-link:metering_integration``. Afterwards run ``osism apply metering``.

File: roles/metering/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+metering_configuration_directory: /opt/metering/configuration
+metering_data_directory: /opt/metering/data
+metering_docker_compose_directory: /opt/metering
+
+metering_tag: "{{ metering_version }}"
+metering_image: "metering:{{ metering_tag }}"
+
+metering_container_name: metering
+
+metering_network: 172.31.100.16/28
```
Comment (berendt) [inline roles/metering/defaults/main.yml:31]: Choose the next free range from https://docs.osism.tech/appendix/tables/service-networks.html.

172.31.101.176/28

File: roles/metering/tasks/container-Debian.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    - "{{ metering_data_directory }}"
+    - "{{ metering_docker_compose_directory }}"
+
+- name: Copy settngs.conf file
+  ansible.builtin.template:
+    src: settings.conf.j2
+    dest: "{{ metering_configuration_directory }}/settings.conf"
+    owner: "{{ operator_user }}"
+    group: "{{ operator_group }}"
+    mode: 0640
```
Comment (berendt) [inline roles/metering/tasks/container-Debian.yml:21]: Is a restart of the service required when changing this file? If yes add a restart handler here.

File: roles/metering/tasks/container-Debian.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    state: directory
+    owner: "{{ operator_user }}"
+    group: "{{ operator_group }}"
+    mode: 0750
+  loop:
+    - "{{ metering_configuration_directory }}"
+    - "{{ metering_data_directory }}"
+    - "{{ metering_docker_compose_directory }}"
+
+- name: Copy settngs.conf file
```
Comment (berendt) [inline roles/metering/tasks/container-Debian.yml:15]: Typo: settings.conf

File: roles/metering/templates/settings.conf.j2 [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    flavor
+
+[output]
+file = pushed_billing_data
+odoo =
+
+[odoo]
+odoo_db =
+odoo_user_name =
+odoo_api_key =
```
Comment (berendt) [inline roles/metering/templates/settings.conf.j2:20]: Add /n here.

File: roles/metering/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+metering_version: xena
+
+##########################
+# container
+
+metering_configuration_directory: /opt/metering/configuration
+metering_data_directory: /opt/metering/data
+metering_docker_compose_directory: /opt/metering
+
+metering_tag: "{{ metering_version }}"
```
Comment (berendt) [inline roles/metering/defaults/main.yml:26]: Why not simply metering_tag: zed here as default? This way we do not have to introduce the metering_version parameter.

File: roles/metering/meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+galaxy_info:
+  author: Heiko Ritter
+  description: Role osism.services.metering
+  company: cloudandheat GmbH
+  license: Apache License 2.0
+  min_ansible_version: 2.10.0
+  platforms:
+    - name: Ubuntu
+      versions:
+        - focal
```
[the remark this answers is not in the document]
Comment (cah-link) [reply roles/metering/meta/main.yml]: removed focal from list

File: roles/metering/defaults/main.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+metering_image: "metering:{{ metering_tag }}"
```
[the remark this answers is not in the document]
Comment (cah-link) [reply roles/metering/defaults/main.yml]: added the quay.io registry

File: roles/metering/defaults/main.yml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+metering_container_name: metering
+
+metering_network: 172.31.100.16/28
+
+metering_service_name: "docker-compose@metering"
+
+
```
[the remark this answers is not in the document]
Comment (cah-link) [reply roles/metering/defaults/main.yml]: done. Thanks!

File: roles/metering/defaults/main.yml (hunk 2 above)
Comment (cah-link) [reply roles/metering/defaults/main.yml]: Thanks!

File: roles/metering/tasks/container-Debian.yml (hunk 4 above)
Comment (cah-link) [reply roles/metering/tasks/container-Debian.yml]: Thanks for the hint!

File: roles/metering/templates/settings.conf.j2 (hunk 5 above)
Comment (cah-link) [reply roles/metering/templates/settings.conf.j2]: Done

File: roles/metering/defaults/main.yml (hunk 6 above)
Comment (cah-link) [reply roles/metering/defaults/main.yml]: Thanks. I set the metering_tag to zed

File: roles/metering/templates/settings.conf.j2 (hunk 5 above)
Comment (osfrickler) [inline roles/metering/templates/settings.conf.j2:20]: Is this really the way the settings should look like? Or do the empty variables need to be filled in somehow? Most likely there should be role variables then that get referenced in the template?

Comment (cah-link) [reply roles/metering/templates/settings.conf.j2:20]: those variables only has to be filled in if you want to use odoo as billing sink. those fields are ignored as long as they are not used.

Comment (berendt) [reply roles/metering/templates/settings.conf.j2:20]: Parameterization we can do in a follow up. This is rather MVP for now.

**Comment by berendt** [comment]
@cah-link Please rebase and add the new documentation to ``doc/source/index.rst``.

**Review by berendt — APPROVED** [review]

**Pull request merged**